src/org/zalando/stups/mint/worker/external/bucket_storage.clj

```diff
@@ -0,0 +1,26 @@
+(ns org.zalando.stups.mint.worker.external.bucket_storage)
+
+(defmacro StorageException
+  [msg data]
+  `(ex-info ~msg (merge ~data {:type "StorageException"})))
+
+(defn storage-exception? [e]
+  (= "StorageException" (:type (ex-data e))))
+
+; TODO: actually infer the bucket type
+(defn infer-bucket-type
+  "infer bucket type from bucket name"
+  [bucket-name]
+  (if (.startsWith bucket-name "gs://") :gs :s3))
+
+(defmulti writable?
+          "Check if a bucket is writable?"
+          (fn [x] (infer-bucket-type (x :bucket-name))))
```

Comment:
I have updated it to use the same signature. I added `&` so it can handle extra args which we will need for the Google storage interface as we talked about yesterday.